ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py

```diff
@@ -508,18 +515,22 @@ def parse_result(self, dsconfs, result):
 
         # Parse values
         self.valuemap = {}
+        db_regex = re.compile('(.*):counter:(.*):value:(.*)')
         for counterline in filter_sql_stdout(result.stdout):
-            key, value = counterline.split(':', 1)
-            if key.strip() == 'databasename':
-                databasename = value.strip()
-                if databasename not in self.valuemap:
-                    self.valuemap[databasename] = {}
-            elif key.strip() == 'ckey':
-                _counter = value.strip().lower()
-            elif key.strip() == 'cvalue':
-                self.valuemap[databasename][_counter] = value.strip()
-            elif key.strip() == 'databasestatus':
+            try:
+                databasename, _counter, value = db_regex.match(counterline).groups()
+                databasename = databasename.strip()
+                _counter = _counter.strip().lower()
+                value = value.strip()
+            except Exception:
+                log.debug('MSSQL parse_result error in data: %s', counterline)
+                continue
+            if databasename not in self.valuemap:
+                self.valuemap[databasename] = {}
+            if _counter.strip() == 'databasestatus':
```

Comment:
do we need `_counter.strip()` ? (we have stripped it during creation `_counter = _counter.strip().lower()` )

Reply:
thanks for catching that.  i was just leaving some of the original logic which had that.  also found we were double stripping the value as well.